fenix/app/src/test/java/org/mozilla/fenix/shopping/store/ReviewQualityCheckStoreTest.kt

```diff
@@ -1282,6 +1282,55 @@ class ReviewQualityCheckStoreTest {
             assertTrue(observedState.contains(expectedProgressState3))
         }
 
+    @Test
+    fun `GIVEN analysis succeeds WHEN deletedProduct is true and deletedProductReported is false THEN product not available is shown`() =
+        runTest {
+            val tested = ReviewQualityCheckStore(
+                middleware = provideReviewQualityCheckMiddleware(
+                    reviewQualityCheckPreferences = FakeReviewQualityCheckPreferences(isEnabled = true),
+                    reviewQualityCheckService = FakeReviewQualityCheckService(
+                        productAnalysis = {
+                            ProductAnalysisTestData.productAnalysis(
+                                deletedProduct = true,
+                                deletedProductReported = false,
+                            )
+                        },
+                        reanalysis = AnalysisStatusDto.PENDING,
+                        statusProgress = {
+                            AnalysisStatusProgressDto(
+                                status = AnalysisStatusDto.COMPLETED,
+                                progress = 100.0,
+                            )
+                        },
+                    ),
+                    networkChecker = FakeNetworkChecker(isConnected = true),
```

Comment:
Let's set report to report created so we know this only should happen when report created is returned. On returning another response, UI should react differently. OOC, what do we show when not deleted is returned?

Reply:
Updated to return REPORT_CREATED for this test, also added a test to return NOT_DELETED which will return no analysis state since that's what desktop is doing

Reply:
Thanks! 

For not deleted, see [this](https://github.com/mozilla-mobile/firefox-android/pull/4756#discussion_r1425158603).